From 01572a2a96eb30775b3f53caa842dd25998cf7a5 Mon Sep 17 00:00:00 2001
From: Cnetwork <shuguang.wang1990@gmail.com>
Date: Mon, 21 Apr 2014 19:51:56 +0800
Subject: [PATCH 05/25] which are perfect squares?

---
 006_waps_7.rb | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 006_waps_7.rb

diff --git a/006_waps_7.rb b/006_waps_7.rb
new file mode 100644
index 0000000..4f10caa
--- /dev/null
+++ b/006_waps_7.rb
@@ -0,0 +1,27 @@
+#Given an array of number, which are perfect squares?
+#this means you choose square numbers from the array
+
+=begin
+pay attention the range is not array! so the array = 1..16 cannot use array[1]
+the array can do map select dircetirly and it can do some other things just like array can
+=end
+
+def gets_squares(array)
+  array.select{ |i| Math.sqrt(i) % 1 == 0}.uniq.sort
+end
+# Math.sqrt() = i**0.5
+
+#my solution :the answer is right but it is totally mass
+#understand wrong not one by one but select
+def get_squares(array)
+  #Enter code here!
+  arr = array.to_a.sort
+  p array
+  number = arr[-1]
+  str = Array.new
+  (1..number).map {|n| (arr.include?(n*n))&&(n*n <= number) ? str<<n*n :{}}
+  str
+end
+
+
+p get_squares(1..2)
-- 
2.5.5

